### PR TITLE
docs: Create Implement Tab on New Docs Site

### PR DESCRIPTION
**Pull request by taylorvnoj**
## Motivations

The Design Tab can get pretty full, and sometimes there are developer specific topics that can live elsewhere. This PR is a PART 1 of splitting concerns.  

We're creating a (currently named) Implement Tab.  So far, all content can be sorted under a `Configuration` or `Developer Notes` heading.  

Note: There will be duplication between Design and Implement Tabs and this is because, we are pulling from Storybook for the Design Tab content and I don't want to remove information from there yet.  Once we are hard launched and the New Site is the main documentation site, we can update those `stories` files.

Screenshot 2024-12-11 at 9 43 58 AM

### Added

-  `site/src/content` component folders got a new `[componentName]Notes.mdx` file.  I decided to name is this way because we aren't set on a Tab name yet and notes encapsulates it in a general way
- `notes?` was added to our `ContentExport`. The Implement Tab will only be shown if a component has a corresponding `[componentName]Notes.mdx` file.
- `ComponentView` has a new Tab object for the Implement Tab

### Changed

- some `stories` files got updates; some grammar changes, some heading changes so all heading are the same per component.
- `FormatTime` `stories` got a bit of an update to provide more clarity

## Testing

Check that the Implement Tab, when present, has content.  If a component (ex Glimmer) doesn't have a `[componentName]Notes.mdx` file, make sure an Implement Tab isn't present.

Changes can be
[tested via Pre-release](https://github.com/GetJobber/atlantis/actions/workflows/trigger-qa-build.yml)

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: packages/site/src/content/AnimatedPresence/AnimatedPresenceNotes.mdx [hunk 1]
```diff
@@ -0,0 +1,40 @@
+## Configuration
```
Comment (scotttjob) [inline packages/site/src/content/AnimatedPresence/AnimatedPresenceNotes.mdx:1]: Do we want to include a comment in the original file(s) for these extractions letting others know that this has technically been duplicated in the short term?

I would hate for us to make a correction in the original docs somewhere and then lose it when we cut out that work later.

File: packages/site/src/content/AnimatedPresence/index.tsx [hunk 2]
```diff
@@ -44,4 +45,5 @@ export default {
       ),
     },
   ],
+  notes: () => <Notes />,
```
Comment (scotttjob) [inline packages/site/src/content/AnimatedPresence/index.tsx:48]: 👍

File: packages/site/src/content/AnimatedSwitcher/index.tsx [hunk 3]
```diff
@@ -38,4 +39,5 @@ export default {
       ),
     },
   ],
+  notes: () => <Notes />,
```
Comment (scotttjob) [inline packages/site/src/content/AnimatedSwitcher/index.tsx:42]: Yup like this pattern of linking MDX files from the content page. Great work. :)

File: packages/site/src/layout/ComponentView.tsx [hunk 4]
```diff
@@ -106,6 +106,14 @@ export const ComponentView = () => {
         </>
       ),
     },
+    {
+      label: "Implement",
```
Comment (scotttjob) [inline packages/site/src/layout/ComponentView.tsx:110]: This looks okay to me and shouldn't conflict too heavily with either of the link PRs I have in flight.

File: packages/site/src/layout/ComponentView.tsx [hunk 5]
```diff
@@ -118,6 +126,10 @@ export const ComponentView = () => {
         return false;
       }
 
+      if (!PageMeta?.notes && index === 3) {
```
Comment (scotttjob) [inline packages/site/src/layout/ComponentView.tsx:129]: This also looks okay. We'll have to adjust the auto-links after we merge, but not a huge lift to do that.

**Review by scotttjob — APPROVED** [review]
LGTM!. Let's work together on the merge, I have a feeling we're going to conflict because we were both working on the index files.

File: packages/site/src/content/AnimatedPresence/AnimatedPresenceNotes.mdx (hunk 1 above)
Comment (taylorvnoj) [reply packages/site/src/content/AnimatedPresence/AnimatedPresenceNotes.mdx:1]: Yes! Great idea! I will do that in my next set of components

**Pull request merged**